diff --git a/dist/index.cjs b/dist/index.cjs
index c08546dc889273513599ff365864743b03116b65..d431047a417eb92ebac3ded0bc0d7d682601c5af 100644
--- a/dist/index.cjs
+++ b/dist/index.cjs
@@ -1044,6 +1044,33 @@ class FrameToIndexedFrame {
     write: (frame) => {
       const transparentIndex = this._config.backgroundColorIndex;
       const pixels = frame.data;
+      // Floyd-Steinberg Dithering for alpha channel
+      const toLinear = x => (x / 255) ** 2.2;
+      const fromLinear = x => Math.round((x ** (1 / 2.2)) * 255);
+      const pos = (x, y, c) => (x * frame.width + y) * 4 + c;
+      for (let y = 0; y < frame.height; y++) {
+        for (let x = 0; x < frame.width; x++) {
+          const hasLeft = x > 1;
+          const hasRight = x + 1 < frame.width;
+          // const hasUp = y > 1;
+          const hasDown = y + 1 < frame.height;
+          const oldAlphaLinear = toLinear(pixels[pos(x, y, 3)]);
+          const newAlphaLinear = oldAlphaLinear > .5 ? 1 : 0;
+          const error = oldAlphaLinear - newAlphaLinear;
+          pixels[pos(x, y, 3)] = fromLinear(newAlphaLinear);
+          for (const [check, [xOffset, yOffset], weight] of [
+            [hasRight, [1, 0], 7 / 16],
+            [hasLeft && hasDown, [-1, 1], 3 / 16],
+            [hasDown, [0, 1], 5 / 16],
+            [hasRight && hasDown, [1, 1], 1 / 16],
+          ]) {
+            if (check) {
+                const p = pos(x + xOffset, y + yOffset, 3);
+                pixels[p] = fromLinear(toLinear(pixels[p]) + error * weight);
+            }
+          }
+        }
+      }
       let transparent = false;
       const indexes = new Uint8ClampedArray(pixels.length / 4);
       for (let len = pixels.length, i = 0; i < len; i += 4) {
diff --git a/dist/index.mjs b/dist/index.mjs
index 648f9f1ec48b0794acff4747d10240e869f63fef..4ec49192758783c97b9d107a9085f37f206b5593 100644
--- a/dist/index.mjs
+++ b/dist/index.mjs
@@ -1042,6 +1042,33 @@ class FrameToIndexedFrame {
     write: (frame) => {
       const transparentIndex = this._config.backgroundColorIndex;
       const pixels = frame.data;
+      // Floyd-Steinberg Dithering for alpha channel
+      const toLinear = x => (x / 255) ** 2.2;
+      const fromLinear = x => Math.round((x ** (1 / 2.2)) * 255);
+      const pos = (x, y, c) => (x * frame.width + y) * 4 + c;
+      for (let y = 0; y < frame.height; y++) {
+        for (let x = 0; x < frame.width; x++) {
+          const hasLeft = x > 1;
+          const hasRight = x + 1 < frame.width;
+          // const hasUp = y > 1;
+          const hasDown = y + 1 < frame.height;
+          const oldAlphaLinear = toLinear(pixels[pos(x, y, 3)]);
+          const newAlphaLinear = oldAlphaLinear > .5 ? 1 : 0;
+          const error = oldAlphaLinear - newAlphaLinear;
+          pixels[pos(x, y, 3)] = fromLinear(newAlphaLinear);
+          for (const [check, [xOffset, yOffset], weight] of [
+            [hasRight, [1, 0], 7 / 16],
+            [hasLeft && hasDown, [-1, 1], 3 / 16],
+            [hasDown, [0, 1], 5 / 16],
+            [hasRight && hasDown, [1, 1], 1 / 16],
+          ]) {
+            if (check) {
+                const p = pos(x + xOffset, y + yOffset, 3);
+                pixels[p] = fromLinear(toLinear(pixels[p]) + error * weight);
+            }
+          }
+        }
+      }
       let transparent = false;
       const indexes = new Uint8ClampedArray(pixels.length / 4);
       for (let len = pixels.length, i = 0; i < len; i += 4) {
